packages/concerto-core/lib/introspect/modelfile.js

```diff
@@ -275,19 +275,20 @@ class ModelFile extends Decorated {
         // Validate all of the types in this model file.
         // Check if names of the declarations are unique.
         const uniqueNames = new Set();
-        this.declarations.forEach(
-            d => {
-                const fqn = d.getFullyQualifiedName();
-                if (!uniqueNames.has(fqn)) {
-                    uniqueNames.add(fqn);
-                } else {
-                    throw new IllegalModelException(
-                        `Duplicate class name ${fqn}`
-                    );
-                }
+        const concertoTypes = new Set(['Concept', 'Asset', 'Transaction', 'Participant', 'Event']);
```

Comment:
What happens when more concertoTypes are created?

Reply:
I checked if there is any utility function in Model.Util which checks if the type is a concerto-type or not . There is no such function currently . But yes it would be better if have one function which manages this and we register them here

Reply:
Let's remove the dependency on `concertoTypes` as we cannot keep track of this list of items.

Reply:
Remove the dependency on the concerto-type. Also updated the breaking test-cases